template/pyproject.toml.jinja

```diff
@@ -1,16 +1,27 @@
 [build-system]
-{%- if backend == "hatch" %}
 requires = [
+{%- if backend == "hatch" %}
   "hatchling",
+{%- elif backend == "setuptools" %}
+  "setuptools",
+{%- endif %}
 {%- if backend == "hatch" and vcs == true %}
   "hatch-vcs",
+{%- elif backend == "setuptools" and vcs == true %}
+  "setuptools_scm[toml]>=8.1.0",
 {%- endif %}
 ]
+{%- if backend == "hatch" %}
 build-backend = "hatchling.build"
+{%- elif backend == "setuptools" %}
+build-backend = "setuptools.build_meta"
 {%- endif %}
 
 [project]
 name = "{{ project_name }}"
+{%- if backend == "setuptools" %}
```

Comment:
also an `and` condition checking if `scm` is *not* selected?

Reply:
I think it is mandatory for both cases, I tried to build wheels with `setuptools` without `scm` and removed the version entry from `pyproject.toml` and this is what I got.
```bash
configuration error: `project` must contain ['version'] properties

ERROR Backend subprocess exited when trying to invoke get_requires_for_build_sdist
```

Reply:
With or without `scm` doesn't seem to matter, if it's `setuptools` the wheels fail to build without the version property in `pyproject.toml` when I tested it locally.

Reply:
Yes, the version has to be present – either hardcoded (static), or as a dynamic attribute through `dynamic = ["version"]`.

Reply:
Yes, that's what I meant, sorry for the confusion. The version should be `dynamic` if `scm` is being used and static (`0.0.1`) when `scm` is not selected.